### ISSUE TITLE
Program shutting down with no flow 

### ISSUE THREAD

**Issue by Goatlips** [petergridge/Irrigation-V5#13]
I finally got your program installed - A few minor issues I will give feedback on later when I have more time.

In summary,  A very impressive and simple to use system,  I think it is going to work perfectly for my application

But I have come across an issue during my initial testing.    I had it working well and then decided to add zone enable switches. as well as a intra zone delay for the pump.  The issues only turned up after that.  (but just be aware,  I only tested it once before that so my experience is very limited

Now when I try to run a zone manually,  It appears to turn on solinod and then turn off solinod before the pump has even turned on ,  or at least not enough time to get water to the flow meter so it can register.   I have attached what logs and information that I have.     I will try and recreate under full logs tomorrow morning.

I am based in Perth,  so if it is unclear,  I have no issues going thru on the phone or direct chat if easier.   

Well done on a great looking program

2023-02-03 06:05:10.797 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration irrigationprogram which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2023-02-03 06:05:14.430 ERROR (MainThread) [custom_components.irrigationprogram.switch] Sensor.time not defined check your configuration
2023-02-03 06:05:50.727 WARNING (MainThread) [custom_components.irrigationprogram.irrigationzone] No flow detected, turning off solenoid to allow program to complete
2023-02-03 06:06:25.775 WARNING (MainThread) [custom_components.irrigationprogram.irrigationzone] No flow detected, turning off solenoid to allow program to complete
2023-02-03 06:06:38.579 WARNING (MainThread) [custom_components.irrigationprogram.irrigationzone] No flow detected, turning off solenoid to allow program to complete
2023-02-04 00:40:05.629 WARNING (MainThread) [custom_components.irrigationprogram.irrigationzone] No flow detected, turning off solenoid to allow program to complete
2023-02-04 00:41:19.192 WARNING (MainThread) [custom_components.irrigationprogram.irrigationzone] No flow detected, turning off solenoid to allow program to complete
2023-02-04 00:43:41.234 WARNING (MainThread) [custom_components.irrigationprogram.irrigationzone] No flow detected, turning off solenoid to allow program to complete
2023-02-04 00:55:56.215 WARNING (MainThread) [custom_components.irrigationprogram.irrigationzone] No flow detected, turning off solenoid to allow program to complete
2023-02-04 00:56:14.691 WARNING (MainThread) [custom_components.irrigationprogram.irrigationzone] No flow detected, turning off solenoid to allow program to complete

![image](https://user-images.githubusercontent.com/102002785/216739685-01b3d754-d150-4ff3-a1ee-3fe411d8ef39.png)

**Comment by Goatlips** [comment]
Sorry -  Forgot to add
running 5.2.0 beta 3

system consists of    3 zones ,  each zone with flow meter and soliniod , run off a pump

**Comment by petergridge** [comment]
Hi @Goatlips 

Perth is my home town, but I have been in NSW for the last 20 odd years, I get home once a year or so though. It is good to hear from an Aussie user. Most contacts have been from Europe and North America.

The run days feature was explicitly to support Perth's restrictions.

I can see the problem with the pump and flow meter implementation. The pump is a process that monitors the zones that use it, so it will not start until the zone starts, so I can understand what is happening. I will introduce a delay before the flow rule kicks in and stops the zone.

The interzone delay will cause the pump to turn off between zones, the pump remains active for only 2 seconds or so after a zone completes to wait for the next zone to start. So I suspect that before you put this in place only the first zone failed to start.

The only other issue I notices is you don't appear to have the time configured. This is required for the program to start at the designated time. One day I may get rid of this dependency, but it just works as it is.
```
sensor:
  - platform: time_date
    display_options:
      - 'time'
      - 'date'
``` 

I will have a look at the code this weekend and let you know when I have an update.

Cheers
Pete

**Comment by petergridge** [comment]
Wow that was an easy fix, V5.2.0-beta4 is up deployed.

Let me know how you go and I will close off this issue.

**Comment by Goatlips** [comment]
Thanks Peter,. Thanks for the nice quick fix. I will deploy and give you
some feedback Sunday or Monday.

I will fix the time issue.   I haven't set it to automatic yet,. Been only
running manually so far.  I have to understand andtrust the process before
I go automatic.

I might have to buy you a beer next time you are in Perth.   You might have
saved me lots of programming time.

Glen
